lib/datadog/statsd.rb

```diff
@@ -86,38 +89,68 @@ def self.VERSION
       VERSION
     end
 
+    ##
+    # Get the current instance of a StatsD client
+    #
+    # @param *args [Array<Object>]
+    #     Any number of arguments passed to the constructor when building a new instance
+    #
+    # @return [Datadog::Statsd]
+    #     An instance of the StatsD client or a new one using the provided arguments
+    def self.current(*args)
```

Comment:
would return the same instance when called with different args

Reply:
Hm, that's true, the blue-print was kind of how `Redis.current` works - so there's only one kind of global instance. So maybe a better idea would be to check on whether `args` are present or not. If args are present, always create an memoize a new instance. If no args are present, return an eventually memoized instance or build a default one.

Reply:
I am open to suggestions or alternate ideas here